Resources/public/Admin.js

```diff
@@ -329,6 +329,11 @@ var Admin = {
 
     setup_collection_buttons: function(subject) {
 
+        // Remove previously attached event(s) which causing function to be called multiple times
+        jQuery(subject).on('click', '.sonata-collection-add', function(event){
+            return false;
+        });
+
```

Comment:
Have you tried the latest jquery version? Maybe it's jquery bug or a timing problem.